esp-mbedtls/Cargo.toml

```diff
@@ -14,15 +14,16 @@ log = { version = "0.4.17", default-features = false }
 enumset = { version = "1", default-features = false }
 embedded-io = { version = "0.6.1" }
 embedded-io-async = { version = "0.6.0", optional = true }
-esp-hal = { version = "0.22.0", optional = true }
+esp-hal = { version = "0.23.0", optional = true }
 # For malloc/free
 # TODO: Replace with `esp-alloc` once `esp-alloc` starts to provide `malloc` and `free` in future
 # ... or switch to our own `mbedtls_malloc/free`
-esp-wifi = { version = "0.11.0", default-features = false, optional = true }
+esp-wifi = { version = "0.12.0", default-features = false, optional = true }
 cfg-if = "1.0.0"
-edge-nal = { version = "0.4.0", optional = true }
+edge-nal = { version = "0.5.0", optional = true }
 critical-section = "1.1.3"
 crypto-bigint = { version = "0.5.3", optional = true, default-features = false, features = ["extra-sizes"] }
+nb = "1.1.0"
```

Comment:
Nit: we can make this an optional dep which is enabled only when the `esp-hal` feature is enabled? We don't need it otherwise?